src/swupd/hash.go

```diff
@@ -109,16 +135,16 @@ func set(out []byte, in int64) {
 // hmacComputeKey returns what should be an ascii string as an array of byte
 // it is really ugly to be compatible with the C implementation. It is not portable
 // as the C version isn't portable.
-// The syscall.Stat_t has been butchered
-func hmacComputeKey(info syscall.Stat_t) []byte {
+// The C.struct_stat has been butchered
+func hmacComputeKey(info C.struct_stat) []byte {
```

Comment:
It might be cleaner to use a dedicated struct for the input of hmacComputeKey instead of passing a "butchered" stat struct. Then the function can also be tested again in isolation.